### ISSUE TITLE
Getting NaN values for loss when doing fine tuning on TUAB and when doing evaluation only, getting -1 for all preds.

### ISSUE THREAD

**Issue by karimelzokm** [935963004/NeuroLM#6]
Hello,

I am attempting to run the code file train_instruction.py on TUAB dataset. I have cloned the GitHub repo directly and ran all the data preparation scripts according to the README.MD. I am also using the pretrained checkpoints NeuroLM-B.pt and VQ.pt published on HuggingFace. The first iteration of the first epoch runs correctly, but then on step 2 of first epoch, it is giving NaNs for the loss, the logits, (and after inspecting a bit, the gradients as well). 

This is a sample output: 

```
epoch 1 step [1/18531]: train total loss 22.9526, instruction loss 19.7728, text loss 3.1798
epoch 1 step [2/18531]: train total loss nan, instruction loss nan, text loss nan
epoch 1 step [3/18531]: train total loss nan, instruction loss nan, text loss nan
```

I have also attempted to run using eval_only from the checkpoint and I am getting -1 for all the predictions. When I inspected the `pred_string` variable, it seems that the answer is just some random characters and some Unicode characters. I have attempted to resolve these issues, but I was unable to come to anything. Did you guys encounter something like this during your runs? It would be great if you can aid me in resolving this! Thanks.

**Comment by 935963004** [comment]
Hi, does your GPU suooort bf16? I think you could try to use float32 in case of overflow.
